airbyte-integrations/bases/bases-destination-jdbc/src/main/java/io/airbyte/integrations/destination/staging/SerialFlush.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.destination.staging;
+
+import static java.util.stream.Collectors.joining;
+
+import com.google.common.annotations.VisibleForTesting;
+import io.airbyte.commons.exceptions.ConfigErrorException;
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.db.jdbc.JdbcDatabase;
+import io.airbyte.integrations.destination.jdbc.WriteConfig;
+import io.airbyte.integrations.destination.record_buffer.FlushBufferFunction;
+import io.airbyte.protocol.models.v0.AirbyteStreamNameNamespacePair;
+import io.airbyte.protocol.models.v0.ConfiguredAirbyteCatalog;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.io.FileUtils;
+
+/**
+ * Serial flushing logic. Though simpler, this causes unnecessary backpressure and slows down the
+ * entire pipeline.
+ * <p>
+ * Note: This class should be re-written so that is implements the {@link FlushBufferFunction}
+ * interface, instead of return an anonymous function implementing this interface for clarity. As of
+ * this writing, we avoid doing so to simplify the migration to async flushing.
+ */
+@Slf4j
+public class SerialFlush {
```

Comment:
where would this be used?

Reply:
`StagingConsumerFactory.create` = the default consumer constructor

Reply:
ah. i see.

Reply:
felt this at least highlighted the difference between the flush function vs the current state of `getYFunction` returning anonymous functions